server/datastore/mysql/labels.go

```diff
@@ -444,12 +443,16 @@ func (ds *Datastore) ListLabels(ctx context.Context, filter fleet.TeamFilter, op
 		return nil, &fleet.BadRequestError{Message: "parameter 'query' is not supported"}
 	}
 
-	query := fmt.Sprintf(`
-			SELECT *,
-				(SELECT COUNT(1) FROM label_membership lm JOIN hosts h ON (lm.host_id = h.id) WHERE label_id = l.id AND %s) AS host_count
-			FROM labels l
-		`, ds.whereFilterHostsByTeams(filter, "h"),
-	)
+	query := "SELECT * FROM labels l "
+	// If a team filter is provided, filter host membership by team and return counts with the labels.
+	if filter.User != nil {
+		query = fmt.Sprintf(`
+				SELECT *,
+					(SELECT COUNT(1) FROM label_membership lm JOIN hosts h ON (lm.host_id = h.id) WHERE label_id = l.id AND %s) AS host_count
+				FROM labels l
+			`, ds.whereFilterHostsByTeams(filter, "h"),
+		)
+	}
```

Comment:
Small optimization to skip running the subquery when we don't have a real team filter.  Currently if the team filter doesn't have a User, it adds `FALSE` to the end of the subquery so that it always returns 0.